### PR TITLE
ci: add workspace unit tests to coverage tests

### PR DESCRIPTION
**Pull request by taimingl**
## Summary by CodeRabbit

- **New Features**
	- Enhanced code coverage testing with the addition of a `--workspace` option.
	- Improved configuration validation for SNS endpoint URLs and minimum data retention days.

- **Bug Fixes**
	- Updated cache removal logic to clarify the impact on accessed keys.

- **Tests**
	- Expanded test cases for configuration checks and cache behavior to ensure robustness and accuracy.

### CODE REVIEW

File: src/infra/src/cache/file_data/mod.rs [hunk 1]
```diff
@@ -220,9 +220,9 @@ mod tests {
         cache.insert(key1.to_string(), 1);
         cache.insert(key2.to_string(), 2);
         cache.contains_key(key1);
-        cache.remove();
-        assert!(cache.contains_key(key1));
-        assert!(!cache.contains_key(key2));
+        cache.remove(); // contains_key() does not mark the key1 as used -> removed
```
Comment (taimingl) [inline src/infra/src/cache/file_data/mod.rs:223]: @haohuaijin please confirm if this is desired/acceptable behavior. The original test case wasn't passing and I figured that was the issue.

File: src/config/src/config.rs [hunk 2]
```diff
@@ -2048,10 +2048,6 @@ mod tests {
         check_sns_config(&mut cfg).unwrap();
         assert_eq!(cfg.sns.endpoint, "https://sns.us-west-2.amazonaws.com");
 
-        // Test invalid endpoint
-        cfg.sns.endpoint = "invalid-url".to_string();
-        assert!(check_sns_config(&mut cfg).is_err());
```
Comment (taimingl) [inline src/config/src/config.rs:2053]: This is tested below.

**Review by patrickcsullivan — APPROVED** [review]
🎉

File: src/infra/src/cache/file_data/mod.rs (hunk 1 above)
Comment (hengfeiyang) [reply src/infra/src/cache/file_data/mod.rs:223]: You are right, at the beginning the LRU contains_key effect the result, but we fixed it😁
https://github.com/kyren/hashlink/issues/27

**Review by hengfeiyang — APPROVED** [review]

**Pull request merged**